internal/providers/errors/errors.go

```diff
@@ -0,0 +1,17 @@
+// Package errors provides custom error types which can be returned by the providers
+package errors
+
+import "fmt"
+
+// ForDisplayError is an error type for errors that are meant to be displayed to the user.
+type ForDisplayError struct {
+	message string
+}
+
+func (e *ForDisplayError) Error() string {
+	return e.message
+}
+
+func NewForDisplayError(format string, v ...interface{}) *ForDisplayError {
```

Comment:
Also, do not return a pointer unless we really need it.

Reply:
done